AxonDeepSeg/apply_model.py

```diff
@@ -464,10 +494,15 @@ def perform_batch_inference(tf_session, tf_prediction_op, tf_input, batch_x, siz
     :return: List of segmentation of the patches, and optionally list of the probabilty maps for each patch.
     """
 
-    p = tf_session.run(tf_prediction_op, feed_dict={
-        tf_input: batch_x})  # here we get the predictions under prob format (float, between 0 and 1, shape = (bs, size_image*size_image, n_classes).
-    Mask = np.argmax(p, axis=2)
-    Mask = Mask.reshape(size_batch, input_size, input_size)  # Now Mask is a 256*256 mask with Mask[i,j] = pixel_class
+    batch_x = np.expand_dims(batch_x, axis=3)
+    batch_x = np.concatenate((batch_x, batch_x, batch_x), axis=3)
+
+    p = model.predict(batch_x)
+
+    """
+          # here we get the predictions under prob format (float, between 0 and 1, shape = (bs, size_image*size_image, n_classes)."""
+    Mask = np.argmax(p, axis=3)  # Now Mask is a 256*256 mask with Mask[i,j] = pixel_class
```

Comment:
is 

```python
Mask = np.argmax(p, axis=3)
```

really equivalent to

```python
Mask = np.argmax(p, axis=2)		
Mask = Mask.reshape(size_batch, input_size, input_size)
```

?

Reply:
Yes both are equivalent